src/Persistence_representations/include/gudhi/PSSK.h

```diff
@@ -5,163 +5,167 @@
+    std::vector<std::pair<double, double> > intervals;
+    if (dimension == std::numeric_limits<unsigned int>::max()) {
+      intervals = read_persistence_intervals_in_one_dimension_from_file(filename);
     } else {
-      intervals_ = read_persistence_intervals_in_one_dimension_from_file(filename, dimension);
+      intervals = read_persistence_intervals_in_one_dimension_from_file(filename, dimension);
     }
-    this->construct(intervals_, filter, number_of_pixels, min_, max_);
+    this->_construct(intervals, filter, number_of_pixels, min, max);
   }
 
- protected:
-  void construct(const std::vector<std::pair<double, double> >& intervals_,
-                 std::vector<std::vector<double> > filter = create_Gaussian_filter(5, 1),
-                 size_t number_of_pixels = 1000,
-                 double min_ = -1,
-                 double max_ = -1);
-};
-
-// if min_ == max_, then the program is requested to set up the values itself based on persistence intervals
-void PSSK::construct(const std::vector<std::pair<double, double> >& intervals_,
-                     std::vector<std::vector<double> > filter,
-                     size_t number_of_pixels,
-                     double min_,
-                     double max_)
-{
-  bool dbg = false;
-  if (dbg) {
-    std::cerr << "Entering construct procedure \n";
-    getchar();
-  }
-
-  if (min_ == max_) {
-    // in this case, we want the program to set up the min_ and max_ values by itself.
-    min_ = std::numeric_limits<int>::max();
-    max_ = -std::numeric_limits<int>::max();
+ private:
+  // if min == max, then the program is requested to set up the values itself based on persistence intervals
+  void _construct(const std::vector<std::pair<double, double> >& intervals,
+                  const std::vector<std::vector<double> >& filter = create_Gaussian_filter(5, 1),
+                  std::size_t number_of_pixels = 1000,
+                  double min = -1,
+                  double max = -1)
+  {
+#ifdef GUDHI_DEBUG
+    std::clog << "Entering construct procedure \n";
+#endif
+
+    if (min == max) {
+      // in this case, we want the program to set up the min and max values by itself.
+      min = std::numeric_limits<int>::max();
+      max = std::numeric_limits<int>::min();
+
+      for (std::size_t i = 0; i != intervals.size(); ++i) {
+        if (intervals[i].first < min) min = intervals[i].first;
+        if (intervals[i].second > max) max = intervals[i].second;
+      }
+      // now we have the structure filled in, and moreover we know min and max values of the interval, so we know the
+      // range.
 
-    for (size_t i = 0; i != intervals_.size(); ++i) {
-      if (intervals_[i].first < min_) min_ = intervals_[i].first;
-      if (intervals_[i].second > max_) max_ = intervals_[i].second;
+      // add some more space:
+      min -= std::fabs(max - min) / 100;
+      max += std::fabs(max - min) / 100;
```

Comment:
It would make more sense to compute `fabs(max-min)/100` once and then add/remove the same quantity from min/max. Adding a bit more space on the right is strange.